### PR TITLE
feat(response): add link-extension to append_link()

### LINKED ISSUE #288: Add support for link-extension to Response.append_link
See also: https://tools.ietf.org/html/rfc5988

### PR DESCRIPTION
**Pull request by kaichan1201**
# Summary of Changes

This PR adds link-extension to append_link() as specified in [RFC 8288 Sec. 3.4.2](https://datatracker.ietf.org/doc/html/rfc8288#section-3.4.2). From my understanding, there is not any restriction on what the extension attributes could be, so I implement this feature in the most intuitive way: having the user pass in pairs of (parameter, value) and concatenate them onto the header.

# Related Issues

Closes #288 

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

### CODE REVIEW

File: falcon/response.py [hunk 1]
```diff
@@ -849,6 +845,9 @@ def append_link(
                 Can take values 'anonymous' or 'use-credentials' or None.
                 (See:
                 https://www.w3.org/TR/html50/infrastructure.html#cors-settings-attribute)
+            link_extension(iterable): Provides additional custom attributes, as
+                described in RFC 8288, Section 3.4.2. Each member of the iterable
+                must be a two-membered tuple in the form of (*param*, *value*).
```
Comment (CaselIT) [inline falcon/response.py:850]:
```suggestion
                must be a two-tuple in the form of (*param*, *value*).
```

also maybe a link to the rfc would not hurt?

File: falcon/response.py [hunk 2]
```diff
@@ -911,6 +910,10 @@ def append_link(
                 # Un-inline in case more values are supported in the future.
                 value += '; crossorigin="use-credentials"'
 
+        if link_extension is not None:
+            value += '; '
+            value += '; '.join([ext[0] + '=' + ext[1] for ext in link_extension])
```
Comment (CaselIT) [inline falcon/response.py:915]: I guess here we could be more rigid by requiring a two-item object, using something like `for p, v in link_extension`

**Review by CaselIT — COMMENTED** [review]
Looks generally ok, just a couple of suggestions

File: falcon/response.py (hunk 1 above)
Comment (kaichan1201) [reply falcon/response.py]: done!

File: falcon/response.py (hunk 2 above)
Comment (kaichan1201) [reply falcon/response.py]: done

**Review by CaselIT — DISMISSED** [review]

**Review by vytas7 — CHANGES_REQUESTED** [review]
LGTM, but before merging, we need to add a Towncrier newsfragment describing the new feature that has been implemented. I unchecked the item in question in the checklist.

**Review by CaselIT — APPROVED** [review]

**Review by vytas7 — APPROVED** [review]
I would like to further polish that newsfragment (e.g., not start with an imperative), but we can too that in a follow-up PR. I noticed that there are more newsfragments that need some cleanup.

**Pull request merged**